### DISCUSSION TITLE
How do I use Image Update Automation with helmreleases?

### DISCUSSION THREAD

**Discussion by gabecazares-cc** [fluxcd/flux2#1481] [Q&A] [answered]
Is image update automation supported on helmreleases? The flux2 docs walk through how to configure image updates on deployments but I'm not sure if helmreleases allow for this as well.

Our helmrelease chart contains a deployment template that I can add the marker to:
` # {"$imagepolicy": "flux-system:podinfo"}`
But flux would then have no deployment manifest to write back to in git since it's a helm chart... And I'm not even sure if the above marker will be picked up by flux since it's not a deployment manifest sitting in the repo and is only created as an artifact of a helmrelease.

Thank you!!

**Comment by stefanprodan** [comment, marked as answer]
Check out the docs: https://fluxcd.io/docs/guides/image-update/#configure-image-update-for-custom-resources

**Reply by gabecazares-cc** [reply]
Thanks! Not sure how I missed that specific example for helmreleases!

I added the markers to my helmrelease file and have been waiting for 30m and not seeing any updates though.

Does this seem right to you?
```
$ flux get image repository
NAME                       	READY	MESSAGE                      	LAST SCAN                	SUSPENDED 
my-app				True 	successful scan, found 6 tags	2021-06-05T06:51:15-06:00	False    	

$ flux get image policy
NAME                       	READY	MESSAGE                                                                                                                        	LATEST IMAGE                                                                                
my-app				True 	Latest image tag for '111111111111.dkr.ecr.us-west-1.amazonaws.com/my-org/my-app' resolved to: 894731d-1622819342		111111111111.dkr.ecr.us-west-1.amazonaws.com/my-org/my-app:894731d-1622819342	

$ cat charts/namespace/my-app.yaml | grep -A2 image:
    image:
      repository: 111111111111.dkr.ecr.us-west-1.amazonaws.com/my-org/my-app # {"$imagepolicy": "flux-system:my-app:name"}
      tag: 41548db # {"$imagepolicy": "flux-system:my-app:tag"}
```
The only thing I can think of is that the existing tag doesn't currently follow the new timestamped tag convention...

**Reply by yebyen** [reply]
Have you created an image automation and pointed it at this path in the GitRepository with write access according to image update guide?

https://fluxcd.io/docs/guides/image-update/#configure-image-updates

ImageUpdateAutomation is the resource which does all of the heavy lifting, ImageRepo and ImagePolicy objects are only part of the picture. You're on the right track, policy and repo auth both look to be in order from what I can see. If you added an ImageUpdateAutomation correctly to enact the ImagePolicy, you can reconcile it or wait for the interval to see the image update behavior in action.

**Reply by gabecazares-cc** [reply]
Thanks! This was the missing piece!
